harness/determined/common/storage/azure_client.py

```diff
@@ -0,0 +1,79 @@
+import logging
+from pathlib import Path
+from typing import List, Optional, Union
+
+from azure.core.exceptions import HttpResponseError, ResourceExistsError
+from azure.storage.blob import BlobServiceClient
+from azure.storage.blob import BlobAnalyticsLogging
+
+from determined.common import util
+
+
+class AzureStorageClient(object):
+    """Connects to an Azure Blob Storage service account."""
+
+    def __init__(
+        self,
+        container: str,
+        connection_string: Optional[str] = None,
+        account_url: Optional[str] = None,
+        credential: Optional[str] = None,
+    ) -> None:
+        if connection_string:
+            self.client = BlobServiceClient.from_connection_string(connection_string)
+        elif account_url:
+            self.client = BlobServiceClient(account_url, credential)
+
+        logging.info("Trying to create Azure Blob Storage Container: {}.".format(container))
+        try:
+            self.client.create_container(container.split("/")[0])
+            logging.info("Successfully created container {}.".format(container))
+        except ResourceExistsError:
+            logging.info(
+                "Container {} already exists, and will be used to store checkpoints.".format(
+                    container
+                )
+            )
+        except HttpResponseError as e:
+            if e.reason == "The requested URI does not represent any resource on the server.":
```

Comment:
any chance there's an "error code" or something? comparing against a human-readable error message is dicey.

Reply:
There is an `error_code` that points to an enum - `azure.storage.blob.StorageErrorCode.invalid_uri`. I'll use that instead.